src/layout/mod.rs

```diff
@@ -1219,6 +1219,14 @@ impl<W: LayoutElement> Layout<W> {
         mon.window_under(pos_within_output)
     }
 
+    // returns a rectangle where the currently focused tile is located (on the active output)
+    pub fn active_tile_visual_rectangle(&mut self) -> Option<Rectangle<i32, Logical>> {
+        let Some(monitor) = self.active_monitor() else {
+            return None;
+        };
+        monitor.active_tile_visual_rectangle()
```

Comment:
What happens if this returns a rectangle out of bounds? For example, a window can be bigger than the workspace, and if it's 2x view size in width, the center will be out of bounds. Or, when you take the workspace switch into account, you can switch focus with keybinds while holding a touchpad workspace switch gesture. I guess some clipping should be done, or just returning None in that case?

Reply:
hopefully fixed by clipping in commit [123c2f0](https://github.com/YaLTeR/niri/pull/245/commits/123c2f040e1891c7ce5d3d7c29d497f48da788cd), however I am not able to test my updated code with multiple monitors/outputs at the moment

Reply:
Tested on dual monitor setup and works as expected